portal-web/test/functional/com/liferay/portalweb/macros/CompanyLogging.macro

```diff
@@ -0,0 +1,89 @@
+definition {
+
+	macro assertCompanyLogIsolated {
+		if ("${portalInstanceName}" == "liferay.com") {
+			var baseCompanyId = JSONCompany.getCompanyId();
+		}
+		else {
+			var baseCompanyId = JSONCompany.getCompanyId(portalInstanceName = "${portalInstanceName}");
+		}
+
+		var logContent = CompanyLogging.downloadCompanyLog(portalInstanceName = "${portalInstanceName}");
+
+		if (!(contains("${logContent}", "companyId=${baseCompanyId}"))) {
+			fail("The ${portalInstanceName} company log file content does not match the expected log content.");
+		}
+
+		var companies = JSONCompany.getAllCompanyNames();
+
+		for (var companyTitle : list "${companies}") {
+			if ("${companyTitle}" != "${portalInstanceName}") {
```

Comment:
Instead of doing extra logic in the for loop, itll be a lot cleaner to just only scan through the companies we need to check instead of going through an extra if statement every time. It is weirder to do in Poshi, you can do ListUtil.remove to achieve this
https://github.com/liferay/liferay-portal/blob/master/modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/util/ListUtil.java